tasks/internal/get-advisory-severity/get-advisory-severity.yaml

```
@@ -119,6 +120,100 @@ spec:
               fi
           }
 
+          # Function to process a batch of CVEs with its own token
+          process_cve_batch() { # Expected arguments are [batch_id, batch_cves]
+              echo "Batch $1: Getting token..."
+              set +x
+              batch_token=$(curl --retry 3 --negotiate -u : "${OSIDB_URL}"/auth/token | jq -r '.access')
+              set -x
+
+              echo "Batch $1: Processing CVEs..."
+
+              while read -r cve; do
+                  if [[ -n "$cve" ]]; then
+                      cache_file="$CVE_CACHE_DIR/${cve}.json"
+
+                      # Note: If we bypass this check in a race condition, the output from OSIDB will be
+                      # the same anyways so it is safe - multiple batches writing identical data is harmless
+                      if [[ -f "$cache_file" ]]; then
+                          continue  # Already processed
+                      fi
+
+                      echo "Batch $1: Processing CVE ${cve}"
+                      curl_url="${OSIDB_URL}/osidb/api/v1/flaws?cve_id=${cve}&include_fields=${INCLUDE_FIELDS}"
+                      set +x
+
+                      # Try API call with current token, refresh and retry if it fails
+                      if ! curl --retry 3 \
+                          -H 'Content-Type: application/json' \
+                          -H "Authorization: Bearer ${batch_token}" \
+                          "$curl_url" | jq '.results[0]' > "$cache_file"; then
+                          echo "Batch $1: API call failed, refreshing token and retrying..."
+                          batch_token=$(curl --retry 3 --negotiate -u : "${OSIDB_URL}"/auth/token | jq -r '.access')
+                          curl --retry 3 -H 'Content-Type: application/json' -H "Authorization: Bearer ${batch_token}" \
+                              "$curl_url" | jq '.results[0]' > "$cache_file"
```

Retry after token refresh doesn't check for failure
```suggestion
                          batch_token=$(curl --retry 3 --negotiate -u : "${OSIDB_URL}"/auth/token | jq -r '.access')
                          # Validate refreshed token
                          if [[ -z "$batch_token" || "$batch_token" == "null" ]]; then
                              echo "Batch $1: ERROR - Failed to refresh authentication token" >&2
                              set -x
                              return 1
                          fi
                          # Retry with refreshed token
                          if ! curl --retry 3 -H 'Content-Type: application/json' -H "Authorization: Bearer ${batch_token}" \
                              "$curl_url" | jq '.results[0]' > "$cache_file"; then
                              echo "Batch $1: ERROR - Failed to fetch CVE ${cve} after token refresh" >&2
                              set -x
                              return 1
                          fi
```

---

Not new to this commit, but this is a great catch. Fixed